zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java

```diff
@@ -283,6 +342,154 @@ public ApiResponse handleApiAction(String name, JSONObject params) throws ApiExc
 
                 removeHistoryReferenceAlerts(rootNode);
             }
+        } else if (ACTION_UPDATE_ALERT.equals(name)) {
+            int alertId = ApiUtils.getIntParam(params, PARAM_ALERT_ID);
+            String alertName = ApiUtils.getNonEmptyStringParam(params, PARAM_ALERT_NAME);
+            int riskId = ApiUtils.getIntParam(params, PARAM_ALERT_RISK);
+            int confidenceId = ApiUtils.getIntParam(params, PARAM_ALERT_CONFIDENCE);
+            String desc = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_DESCRIPTION);
+            String param = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_PARAM);
+            String attack = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_ATTACK);
+            String otherInfo = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_OTHERINFO);
+            String solution = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_SOLUTION);
+            String refs = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_REFS);
+            String evidence = ApiUtils.getOptionalStringParam(params, PARAM_ALERT_EVIDENCE);
+            int cweId = ApiUtils.getIntParam(params, PARAM_ALERT_CWEID);
+            int wascId = ApiUtils.getIntParam(params, PARAM_ALERT_WASCID);
+
+            if (riskId < Alert.RISK_INFO || riskId > Alert.RISK_HIGH) {
+                throwInvalidAlertElementId(
+                        Constant.messages.getString(
+                                "alert.api.action.updateAlert.param.riskId.illegal",
```

Comment:
We don't usually i18n the exception messages in the API.

Reply:
I’m not set on this, I just figured since all the descriptions etc were being added I could also tackle this....

Reply:
Right, I'm fine with that.